synapse/types.py

```diff
@@ -214,7 +214,8 @@ def from_string(cls, s):
         return group_id
 
 
-mxid_localpart_allowed_characters = set("_-./=" + string.ascii_lowercase + string.digits)
+# A regex that matches any valid mxid characters
+MXID_LOCALPART_REGEX = re.compile("^[_\-./=a-z0-9]*$")
```

Comment:
As of https://www.regular-expressions.info/anchors.html you might want to prefer
`"\A[_\-./=a-z0-9]*\Z"`

Reply:
That's a good point. I hadn't seen them used before. In my head `^$` were always the start of the string, not the line. The thing I did to solve that used `fullmatch()` instead, which works too. Not sure what is cleaner/faster.